generator/internal/stub_factory_generator.cc

```diff
@@ -95,24 +98,38 @@ Status StubFactoryGenerator::GenerateCc() {
 std::shared_ptr<$stub_class_name$>
 CreateDefault$stub_class_name$(
     google::cloud::CompletionQueue cq, Options const& options) {
-  auto auth = google::cloud::internal::CreateAuthenticationStrategy(
-      std::move(cq), options);
-  auto channel = auth->CreateChannel(
-    options.get<EndpointOption>(), internal::MakeChannelArguments(options));
-  auto service_grpc_stub = $grpc_stub_fqn$::NewStub(channel);)""");
+  auto endpoint = internal::DetermineServiceEndpoint(
+      internal::GetEnv("$service_endpoint_env_var$"),
+      internal::FetchOption<EndpointOption>(options), "$service_endpoint$",
+      options);
+
+  std::shared_ptr<$stub_class_name$> stub;
+  std::shared_ptr<internal::GrpcAuthenticationStrategy> auth;
+  if (!endpoint.ok()) {
+    Options error_options = options;
+    error_options.set<google::cloud::UnifiedCredentialsOption>(
+        internal::MakeErrorCredentials(endpoint.status()));
+    auth = internal::CreateAuthenticationStrategy(CompletionQueue{},
+                                                  error_options);
+  } else {
+    auth = google::cloud::internal::CreateAuthenticationStrategy(std::move(cq),
+                                                                 options);
+    auto channel = auth->CreateChannel(
+      *endpoint, internal::MakeChannelArguments(options));
+    auto service_grpc_stub = $grpc_stub_fqn$::NewStub(channel);)""");
 
   if (!HasLongrunningMethod()) {
     CcPrint(R"""(
-  std::shared_ptr<$stub_class_name$> stub =
-    std::make_shared<Default$stub_class_name$>(std::move(service_grpc_stub));
-)""");
+    stub =
+      std::make_shared<Default$stub_class_name$>(std::move(service_grpc_stub));
+  })""");
   } else {
     CcPrint(R"""(
-  std::shared_ptr<$stub_class_name$> stub =
-    std::make_shared<Default$stub_class_name$>(
-      std::move(service_grpc_stub),
-      google::longrunning::Operations::NewStub(channel));
-)""");
+    stub =
+      std::make_shared<Default$stub_class_name$>(
+        std::move(service_grpc_stub),
+        google::longrunning::Operations::NewStub(channel));
+  })""");
   }
   CcPrint(R"""(
```

Comment:
It looks like `stub` could be null here (in the `!endpoint.ok()` case), which appears problematic.